test/engine/test_pool.py

```diff
@@ -866,6 +868,17 @@ def test_timeout(self):
         assert_raises(tsa.exc.TimeoutError, p.connect)
         assert int(time.time() - now) == 2
 
+    @testing.requires.timing_intensive
+    def test_timeout_subsecond_precision(self):
+        p = self._queuepool_fixture(pool_size=1, max_overflow=0, timeout=0.5)
+        c1 = p.connect()  # noqa
+        with pytest.raises(tsa.exc.TimeoutError, match=r".* timeout 0.50 .*"):
```

Comment:
**mike bayer** (zzzeek) wrote:

we have a context manage testing.expect_raises() to use here.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2373

Reply:
**Gord Thompson** (gordthompson) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2373